api.planx.uk/lib/notify/index.ts

```diff
@@ -65,10 +28,17 @@ const sendEmail = async (
       message: string;
       expiryDate?: string;
     } = { message: "Success" };
-    if (template === "expiry")
-      await softDeleteSession(config.personalisation.sessionId!);
-    if (template === "save")
-      returnValue.expiryDate = config.personalisation.expiryDate;
+
+    if (template === "expiry") {
+      const expiryConfig = config as TemplateRegistry["expiry"]["config"];
+      await softDeleteSession(expiryConfig.personalisation.sessionId);
+    }
+
+    if (template === "save") {
+      const saveConfig = config as TemplateRegistry["save"]["config"];
+      returnValue.expiryDate = saveConfig.personalisation.expiryDate;
+    }
```

Comment:
Casting is necessary here, TS is unable to match and type narrow on the shared generic `T`.

Same problem as previously discussed here - https://github.com/theopensystemslab/planx-new/pull/4153#discussion_r1919809762